bool enabled
float32 distance